edge-modules/MetricsCollector/Settings.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public class Settings
+    {
+        static readonly Lazy<Settings> DefaultSettings = new Lazy<Settings>(
+            () =>
+            {
+                IConfiguration configuration = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("config/settings.json")
+                    .AddEnvironmentVariables()
+                    .Build();
+
+                return new Settings(
+                    configuration.GetValue<string>("AzMonWorkspaceId"),
+                    configuration.GetValue<string>("AzMonWorkspaceKey"),
+                    configuration.GetValue<string>("AzMonLogType"),
+                    configuration.GetValue<string>("MetricsEndpointsCSV"),
+                    configuration.GetValue<int>("ScrapeFrequencyInSecs"),
+                    configuration.GetValue<UploadTarget>("UploadTarget"));
+                });
+
+        Settings(
+            string azMonWorkspaceId,
+            string azMonWorkspaceKey,
+            string azMonLogType,
+            string endpoints,
+            int scrapeFrequencySecs,
+            UploadTarget uploadTarget)
+        {
+            this.AzMonWorkspaceId = Preconditions.CheckNonWhiteSpace(azMonWorkspaceId, nameof(azMonWorkspaceId));
+            this.AzMonWorkspaceKey = Preconditions.CheckNonWhiteSpace(azMonWorkspaceKey, nameof(azMonWorkspaceKey));
+            this.AzMonLogType = Preconditions.CheckNonWhiteSpace(azMonLogType, nameof(azMonLogType));
+
+            this.Endpoints = new List<string>();
+            foreach (string endpoint in endpoints.Split(","))
+            {
+                if (!string.IsNullOrWhiteSpace(endpoint))
+                {
+                    this.Endpoints.Add(endpoint);
+                }
+            }
+
+            this.ScrapeFrequencySecs = Preconditions.CheckRange(scrapeFrequencySecs, 0);
+            this.UploadTarget = Preconditions.CheckNotNull(uploadTarget);
```

Comment:
no null is needed for enum

Reply:
Are there any checks you recommend here?